lib/player.js

```diff
@@ -1707,6 +1707,57 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     return preloadManager;
   }
 
+  /**
+   * Detach the player from the current media element, if any, and returns a
+   * PreloadManager that contains the loaded manifest of that asset, if any.
+   * Allows for the asset to be re-loaded by this player faster, in the future.
+   * When in src= mode, this detach but does not make a PreloadManager.
+   * Leaves the player in a state where it cannot play media, until it has been
+   * attached to something else.
+   *
+   * @param {boolean=} keepAdManager
+   * @return {!Promise.<?shaka.media.PreloadManager>}
+   * @export
+   */
+  async detachAndSavePreload(keepAdManager = false) {
+    let preloadManager = null;
+    if (this.manifest_ && this.parser_ && this.parserFactory_ &&
```

Comment:
Wouldn't it be easier to just do this:

```js
const preloadManager = await this.unloadAndSavePreload(
    /* initializeMediaSource= */ false, keepAdManager);
await this.detach(keepAdManager);
return preloadManager;
```

The first step of detaching is unloading, so it will have the same effect with far less code.

Reply:
It is less code yes, but in my use tests, this makes the function slower. In Tizen it is around 200ms slower. And we want this function for https://github.com/shaka-project/shaka-player/pull/6575 where the transition should be as fast as possible, also those 200ms make it sometimes not work....

Reply:
We can make an internal function that gets the current preloadmanager and that function is used in this.unloadAndSavePreload and detachAndSavePreload, do you agree with that?

Reply:
That is acceptable.

Reply:
Done!